server/bundler/loader.js

```diff
@@ -101,8 +108,25 @@ function splitTemplate( path, section ) {
 	return result.join( '\n' );
 }
 
-function requireTemplate( module ) {
-	return 'require( ' + JSON.stringify( module ) + ' )( controller.clientRouter );\n';
+function requireTemplate( path, section ) {
+	var regex, result;
+
+	if ( path === '/' ) {
+		path = JSON.stringify( path );
+	} else {
+		regex = utils.pathToRegExp( path );
+		path = '/' + regex.toString().slice( 1, -1 ) + '/';
+	}
+
+	result = [
+		'page( ' + path + ', function( context, next ) {',
+		'	controller.setSection( ' + JSON.stringify( section ) + ' )( context );',
+		'	next();',
+		'} );\n',
+		'require( ' + JSON.stringify( section.module ) + ' )( controller.clientRouter );\n\n'
```

Comment:
This will now be called once for each `path` -- so shouldn't we move that `forEach` loop into this function instead, and only have the `setSection` related routing in that loop?